Automattic/jetpack
# Protect: Add/apply shared AdminSectionHero component

## Description
Display scan results and history while in scanning, error, or idle state.

## Proposed changes:

* Adds and uses a shared `AdminSectionHero` component for Scan, History and Firewall routes

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1451

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch and start Jurassic Tube
* Add plugins/themes with known vulns
* Activate Protect and allow the initial scan to process them
* Review the UI and ensure you can access scan results while the scan is in progress
* Upgrade your plan and allow scan to detect threats
* Review the UI and ensure you can access scan results while the scan is in progress
* Force an error and ensure the error state displays correctly in the header
* Ensure header details for the threats list are accurate
* Ensure no regressions are introduced and that the Firewall header remains unaffected
* Verify that designs are responsive

## Screenshots
Scan Results:
   Desktop:
![Screen Shot on 2024-10-09 at 18-35-01](https://github.com/user-attachments/assets/d13b9b29-0532-4d65-89fe-2dcf9f6fc028)

   Mobile:
![Screen Shot on 2024-10-09 at 18-35-30](https://github.com/user-attachments/assets/9e6c7cc1-084e-4d13-ac79-e63a103800ad)

Scan Results - Error:
   Desktop:
![Screen Shot 2024-10-09 at 18 18 38](https://github.com/user-attachments/assets/434034f3-7175-49fa-b80b-4a57a90f64f9)

   Mobile:
![Screen Shot on 2024-10-09 at 18-18-58](https://github.com/user-attachments/assets/445fdca0-4168-4b96-810a-b2516c303177)

Scan Results - Scanning:
   Desktop:
![Screen Shot on 2024-10-09 at 18-28-25](https://github.com/user-attachments/assets/d29f19e1-d9eb-4dbc-ad10-750fa5d16fae)

   Mobile:
![Screen Shot on 2024-10-09 at 18-29-51](https://github.com/user-attachments/assets/f5c51ac5-ae73-4b8b-b61c-c5e24c3f5b69)

History:
   Desktop:
![Screen Shot on 2024-10-09 at 18-20-04](https://github.com/user-attachments/assets/be2b10fc-ac35-4a9d-8250-77c2ccf0a230)

   Mobile:
![Screen Shot on 2024-10-09 at 18-20-38](https://github.com/user-attachments/assets/16cb85c7-4e6e-46d8-861d-b030cb90fa8f)

History - Error:
   Desktop:
![Screen Shot on 2024-10-09 at 18-22-28](https://github.com/user-attachments/assets/c50b84dd-d2fa-49e4-9e15-77bda5f1ecae)

   Mobile:
![Screen Shot on 2024-10-09 at 18-22-51](https://github.com/user-attachments/assets/f167a6db-5e80-4d46-b1c8-e3178e486d70)

*Firewall now implements the Header component, but views remain generally unchanged
